src/radix_connect/p2p_links/radix_connect_purpose.rs

```diff
@@ -0,0 +1,136 @@
+use crate::prelude::*;
+
+#[derive(
+    Serialize,
+    Deserialize,
+    Clone,
+    Copy,
+    Debug,
+    PartialEq,
+    Eq,
+    Hash,
+    PartialOrd,
+    Ord,
+    uniffi::Enum,
+)]
+pub enum RadixConnectPurpose {
+    #[serde(rename = "general")]
+    General,
+
+    #[serde(rename = "unknown")]
+    Unknown,
+}
+
+impl FromStr for RadixConnectPurpose {
+    type Err = CommonError;
+    fn from_str(s: &str) -> Result<Self> {
+        Self::new_from_json_string(s).map_err(|_| {
+            CommonError::InvalidRadixConnectPurpose {
+                bad_value: s.to_owned(),
+            }
+        })
+    }
+}
+
+impl RadixConnectPurpose {
+    pub fn from_str_default_value(s: &str) -> Self {
+        let result = RadixConnectPurpose::from_str(s);
+
+        match result {
+            Ok(value) => value,
+            Err(_) => RadixConnectPurpose::Unknown,
+        }
+    }
+}
+
+impl std::fmt::Display for RadixConnectPurpose {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.to_json_string())
+    }
+}
+
+impl JsonStringSerializing for RadixConnectPurpose {} // to raw String
+impl JsonStringDeserializing for RadixConnectPurpose {} // from raw String
+
+#[uniffi::export]
+pub fn new_radix_connect_purpose_from_string(
+    string: String,
+) -> RadixConnectPurpose {
+    RadixConnectPurpose::from_str_default_value(&string)
+}
+
+#[uniffi::export]
+pub fn radix_connect_purpose_to_string(kind: RadixConnectPurpose) -> String {
```

Comment:
Why is this API needed? I mean where does the Wallet need the String representation of the purpose?

Reply:
You're right, it's not needed in the Wallet.